### PR TITLE
Improvements to Crypto section at docs.rs

### PR DESCRIPTION
**Pull request by MartaDubno**

### CODE REVIEW

File: exonum/src/crypto.rs [hunk 1]
```diff
@@ -143,13 +181,14 @@ pub trait CryptoHash {
...
+/// Initializes the sodium library and automatically selects faster versions
+/// of the primitives if possible.
 ///
-/// # Panics
+/// # Panic
 ///
 /// Panics if sodium initialization is failed.
 ///
-/// # Examples
+/// # Example
```
Comment (DarkEld3r) [inline exonum/src/crypto.rs]: Plural form is used in the Rust standard library, so I prefer to follow that convention.

File: exonum/src/crypto.rs [hunk 2]
```diff
@@ -143,13 +181,14 @@ pub trait CryptoHash {
     fn hash(&self) -> Hash;
 }
 
-/// Initializes the sodium library and chooses faster versions of the primitives if possible.
+/// Initializes the sodium library and automatically selects faster versions
+/// of the primitives if possible.
 ///
-/// # Panics
+/// # Panic
```
Comment (DarkEld3r) [inline exonum/src/crypto.rs]: Same reason as with "examples". :upside_down_face:

File: exonum/src/crypto.rs [hunk 3]
```diff
@@ -261,16 +316,16 @@ macro_rules! implement_public_sodium_wrapper {
...
 
         /// Creates a new instance from bytes slice.
-        pub fn from_slice(bs: &[u8]) -> Option<Self> {
-            $name_from::from_slice(bs).map($name)
+        pub fn from_slice(bytes_slice: &[u8]) -> Option<Self> {
+            $name_from::from_slice(bytes_slice).map($name)
         }
 
-        /// Returns the hex representation of the binary data.
+        /// Returns the hex representation of binary data.
```
Comment (Maria-Nosyk) [inline exonum/src/crypto.rs]: > the hex representation

a hex

File: exonum/src/crypto.rs [hunk 4]
```diff
@@ -324,16 +379,16 @@ macro_rules! implement_private_sodium_wrapper {
...
```
Comment (Maria-Nosyk) [inline exonum/src/crypto.rs]: same as above

File: exonum/src/crypto.rs [hunk 5]
```diff
@@ -366,6 +421,22 @@ macro_rules! implement_private_sodium_wrapper {
...
 /// Ed25519 public key used to verify digital signatures.
 ///
+/// In public-key cryptography, the system uses a a mathematically related pair
+/// of keys: a public key, which is openly distributed, and a private key,
+/// which should remain confidential. The public key is used to encrypt a
+/// message, which can only be decrypted using the corresponding private key.
+/// The private key is also used for digital signatures over messages.
+/// Knowing the sender's public key, the recipient can check  whether the
+/// message was made by the owner of the corresponding private key. If the
+/// message was changed by a third party, the verification will fail.
```
Comment (Maria-Nosyk) [inline exonum/src/crypto.rs]: > message was changed by a third party,

message is changed

File: exonum/src/crypto.rs [hunk 6]
```diff
@@ -378,7 +449,23 @@ implement_public_sodium_wrapper! {
...
 
 implement_private_sodium_wrapper! {
-/// Ed25519 secret key used to create digital signatures over messages.
+/// Ed25519 private key used to create digital signatures over messages.
+///
+/// In public-key cryptography, the system uses a a mathematically related pair
+/// of keys: a public key, which is openly distributed, and a private key,
+/// which should remain confidential. Data encrypted with a public key can be
+/// decrypted only using the corresponding private key. The sender signs a
+/// message using their private key. Knowing the sender's public key, the
```
Comment (Maria-Nosyk) [inline exonum/src/crypto.rs]: > using their private key

usign his

File: exonum/src/crypto.rs [hunk 7]
```diff
@@ -378,7 +449,23 @@ implement_public_sodium_wrapper! {
...
+/// recipient can check  whether the message was made by the owner of the
+/// corresponding private key. If the message was changed by a third party,
```
Comment (Maria-Nosyk) [inline exonum/src/crypto.rs]: > If the message was changed 

is changed

File: exonum/src/crypto.rs [hunk 8]
```diff
@@ -392,7 +479,12 @@ implement_private_sodium_wrapper! {
 }
 
 implement_public_sodium_wrapper! {
-/// SHA-256 hash.
+/// The given structure is the result of applying the SHA-256 hash function to
+/// data. This function splits the input data into blocks and runs each block
+/// though a cycle of 64 iterations. The result of the function is a hash sum
```
Comment (Maria-Nosyk) [inline exonum/src/crypto.rs]: > though a cycle

through

File: exonum/src/crypto.rs [hunk 9]
```diff
@@ -407,7 +499,17 @@ implement_public_sodium_wrapper! {
...
+/// using a private key. Later it is possible to verify, using the corresponding
+/// public key, that the data have indeed been signed with that private key.
+///
+/// Ed25519 is a signature system that ensures fast signing and key generation,
+/// as well as security and collision resilience.
+///
+/// The example below generates a pair of random public and private keys,
+/// adds certain data, signs the data using the private key and verifies,
+/// using the signature, data and corresponding public key, that the data have
+/// been signed with that public key.
```
Comment (Maria-Nosyk) [inline exonum/src/crypto.rs]: > and verifies, using the signature, data and corresponding public key, that the data have been signed with that public key.

and verifies that the data have been signed with that private key.

File: exonum/src/crypto.rs [hunk 10]
```diff
@@ -424,7 +526,19 @@ implement_public_sodium_wrapper! {
 }
 
 implement_private_sodium_wrapper! {
-/// Ed25519 seed that can be used for deterministic keypair generation.
+/// Ed25519 seed that can be used for deterministic keypair generation. The
+/// seed is a succession of bytes which can be used to generate a certain
+/// keypair. If the same seed is indicated in the generator multiple times,
```
Comment (Maria-Nosyk) [inline exonum/src/crypto.rs]: > +/// Ed25519 seed that can be used for deterministic keypair generation. The seed is a succession of bytes which can be used to generate a certain keypair.

Let's unite these sentences into one: "Ed25519 seed representing a succession of bytes that can be used for deterministic keypair generation.

File: exonum/src/crypto.rs [hunk 11]
```diff
@@ -424,7 +526,19 @@ implement_public_sodium_wrapper! {
...
+/// the generated keys will be the same each time.
+///
+/// Note that this is not the seed added to Exonum transactions for additional
+/// security, this is a separate entity. This structure is useful for testing,
+/// to receive repeatable results. The seed in this structure is either set
+/// manually or selected using the methods below.
+///
+/// The example below generates a pair of a public and private keys taking
```
Comment (Maria-Nosyk) [inline exonum/src/crypto.rs]: > a pair of a public and private keys

a pair of public and private keys

File: exonum/src/crypto.rs [hunk 12]
```diff
@@ -424,7 +526,19 @@ implement_public_sodium_wrapper! {
...
+/// into account the selected seed. The same seed will always lead to the
+/// generation of the same keypair.
```
Comment (Maria-Nosyk) [inline exonum/src/crypto.rs]: > lead to the generation

lead to generation

**Pull request merged**